### PR TITLE
install jupyter-widgets and lckr-jupyterlab-variableinstapector

### PR DESCRIPTION
**Pull request by fenglin-ssb**

### CODE REVIEW

File: docker/jupyterlab/Dockerfile [hunk 1]
```diff
@@ -125,6 +125,11 @@ RUN echo "**** install jupyterlab-git ****" && \
     python3 -m pip install jupyterlab-system-monitor && \
     echo "**** install jupytext ****" && \
     python3 -m pip install jupytext && \
+    #Install jupyter-widgets
+    echo "**** install jupyter-widgets ****" && \
+    jupyter labextension install @jupyter-widgets/jupyterlab-manager && \
```
Comment (bjornandre) [inline docker/jupyterlab/Dockerfile]: This is not necessary, since jupyter-widgets is already installed

Comment (fenglin-ssb) [reply docker/jupyterlab/Dockerfile]: then the extension is having problem. i saw the following error logs:


[W 2022-06-27 11:51:42.109 SingleUserLabApp log:189[] 404 GET /user/fnh@ssb.no/lab/extensions/ipyaggrid/static/remoteEntry.a46b34032dbf77d77343.js (fnh@ssb.no@::ffff:127.0.0.6) 2.42ms
[I 2022-06-27 11:51:45.341 LabApp[] Build is up to date
[IPKernelApp[] ERROR | No such comm target registered: jupyter.widget.control
[IPKernelApp[] WARNING | No such comm: 2b951f29-3a61-479c-a2d2-a11b755b2e70
Picked up JAVA_TOOL_OPTIONS: -XX:+TieredCompilation  -XX:TieredStopAtLevel=1  -XX:+IgnoreUnrecognizedVMOptions  -XX:+UseContainerSupport  -Xquickstart  -server
Picked up JAVA_TOOL_OPTIONS: -XX:+TieredCompilation  -XX:TieredStopAtLevel=1  -XX:+IgnoreUnrecognizedVMOptions  -XX:+UseContainerSupport  -Xquickstart  -server
Picked up JAVA_TOOL_OPTIONS: -XX:+TieredCompilation  -XX:TieredStopAtLevel=1  -XX:+IgnoreUnrecognizedVMOptions  -XX:+UseContainerSupport  -Xquickstart  -server
Picked up JAVA_TOOL_OPTIONS: -XX:+TieredCompilation  -XX:TieredStopAtLevel=1  -XX:+IgnoreUnrecognizedVMOptions  -XX:+UseContainerSupport  -Xquickstart  -server
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/usr/local/spark-3.2.1-bin-hadoop3.2/jars/spark-unsafe_2.12-3.2.1.jar) to constructor java.nio.DirectByteBuffer(long,int)
WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/usr/local/spark-3.2.1-bin-hadoop3.2/jars/spark-unsafe_2.12-3.2.1.jar) to constructor java.nio.DirectByteBuffer(long,int)
WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
22/06/27 11:51:52 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
22/06/27 11:51:52 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Using Spark's default log4j profile: org/apache/spark/log4j-defaults.propertiesUsing Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
22/06/27 11:51:53 WARN Utils: Service 'sparkDriver' could not bind on port 35600. Attempting port 35601.
22/06/27 11:51:53 WARN Utils: Service 'sparkDriver' could not bind on port 35600. Attempting port 35601.
22/06/27 11:51:53 WARN Utils: Service 'sparkDriver' could not bind on port 35601. Attempting port 35602.
22/06/27 11:51:53 WARN Utils: Service 'sparkDriver' could not bind on port 35601. Attempting port 35602.
22/06/27 11:51:53 WARN Utils: Service 'sparkDriver' could not bind on port 35602. Attempting port 35603.
22/06/27 11:51:53 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
22/06/27 11:51:53 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
22/06/27 11:51:53 WARN Utils: Service 'SparkUI' could not bind on port 4042. Attempting port 4043.
22/06/27 11:51:53 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
22/06/27 11:51:53 WARN Utils: Service 'SparkUI' could not bind on port 4041. Attempting port 4042.
22/06/27 11:51:53 WARN Utils: Service 'SparkUI' could not bind on port 4042. Attempting port 4043.
22/06/27 11:51:53 WARN Utils: Service 'SparkUI' could not bind on port 4043. Attempting port 4044.
22/06/27 11:51:56 WARN Utils: Service 'org.apache.spark.network.netty.NettyBlockTransferService' could not bind on port 45600. Attempting port 45601.
22/06/27 11:51:56 WARN Utils: Service 'org.apache.spark.network.netty.NettyBlockTransferService' could not bind on port 45601. Attempting port 45602.
22/06/27 11:51:56 WARN Utils: Service 'org.apache.spark.network.netty.NettyBlockTransferService' could not bind on port 45600. Attempting port 45601.
22/06/27 11:51:56 WARN Utils: Service 'org.apache.spark.network.netty.NettyBlockTransferService' could not bind on port 45601. Attempting port 45602.
22/06/27 11:51:56 WARN Utils: Service 'org.apache.spark.network.netty.NettyBlockTransferService' could not bind on port 45602. Attempting port 45603.
22/06/27 11:51:56 WARN ExecutorAllocationManager: Dynamic allocation without a shuffle service is an experimental feature.
22/06/27 11:51:56 WARN ExecutorAllocationManager: Dynamic allocation without a shuffle service is an experimental feature.
[IPKernelApp[] ERROR | No such comm target registered: jupyter.widget.control
[IPKernelApp[] WARNING | No such comm: 91e63a2b-7229-479e-9937-adb982911bab
[IPKernelApp[] ERROR | No such comm target registered: jupyter.widget.control
[IPKernelApp[] WARNING | No such comm: 2076a30f-5c08-4d7a-a78c-63b2dddcafa7
[IPKernelApp[] ERROR | No such comm target registered: jupyter.widget.control
[IPKernelApp[] WARNING | No such comm: 0fc9a166-5bd3-481a-a74b-86f7a765b000
[W 2022-06-27 11:52:18.273 SingleUserLabApp kernelmanager:628[] Culling 'idle' kernel 'pyspark_k8s' (01c695bb-2750-40dd-9e75-200c338b266a) with 0 connections due to 2614 seconds of inactivity.
22/06/27 11:52:18 WARN JavaUtils: Attempt to delete using native Unix OS command failed for path = /tmp/spark-21bf3264-d010-4e57-88c9-dc019c046c70/pyspark-acc77e1d-18d4-40ff-8859-55d64c749afc. Falling back to Java IO way
java.io.IOException: Failed to delete: /tmp/spark-21bf3264-d010-4e57-88c9-dc019c046c70/pyspark-acc77e1d-18d4-40ff-8859-55d64c749afc
    at org.apache.spark.network.util.JavaUtils.deleteRecursivelyUsingUnixNative(JavaUtils.java:171)
    at org.apache.spark.network.util.JavaUtils.deleteRecursively(JavaUtils.java:110)
    at org.apache.spark.network.util.JavaUtils.deleteRecursively(JavaUtils.java:91)
    at org.apache.spark.util.Utils$.deleteRecursively(Utils.scala:1164)
    at org.apache.spark.util.ShutdownHookManager$.$anonfun$new$4(ShutdownHookManager.scala:65)
    at org.apache.spark.util.ShutdownHookManager$.$anonfun$new$4$adapted(ShutdownHookManager.scala:62)
    at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
    at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
    at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
    at org.apache.spark.util.ShutdownHookManager$.$anonfun$new$2(ShutdownHookManager.scala:62)
    at org.apache.spark.util.SparkShutdownHook.run(ShutdownHookManager.scala:214)
    at org.apache.spark.util.SparkShutdownHookManager.$anonfun$runAll$2(ShutdownHookManager.scala:188)
    at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
    at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2019)
    at org.apache.spark.util.SparkShutdownHookManager.$anonfun$runAll$1(ShutdownHookManager.scala:188)
    at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
    at scala.util.Try$.apply(Try.scala:213)
    at org.apache.spark.util.SparkShutdownHookManager.runAll(ShutdownHookManager.scala:188)
    at org.apache.spark.util.SparkShutdownHookManager$$anon$2.run(ShutdownHookManager.scala:178)
    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
    at java.base/java.lang.Thread.run(Thread.java:829)
22/06/27 11:52:18 WARN ExecutorPodsWatchSnapshotSource: Kubernetes client has been closed.
--- Logging error ---
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/logging/__init__.py", line 1103, in emit
    stream.write(msg + self.terminator)
  File "/opt/conda/lib/python3.10/site-packages/ipykernel/iostream.py", line 555, in write
    self._schedule_flush()
  File "/opt/conda/lib/python3.10/site-packages/ipykernel/iostream.py", line 461, in _schedule_flush
    self.pub_thread.schedule(_schedule_in_thread)
  File "/opt/conda/lib/python3.10/site-packages/ipykernel/iostream.py", line 212, in schedule
    f()
  File "/opt/conda/lib/python3.10/site-packages/ipykernel/iostream.py", line 459, in _schedule_in_thread
    self._io_loop.call_later(self.flush_interval, self._flush)
  File "/opt/conda/lib/python3.10/site-packages/tornado/ioloop.py", line 600, in call_later
    return self.call_at(self.time() + delay, callback, *args, **kwargs)
  File "/opt/conda/lib/python3.10/site-packages/tornado/platform/asyncio.py", line 212, in call_at
    return self.asyncio_loop.call_later(
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 718, in call_later
    timer = self.call_at(self.time() + delay, callback, *args,
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 729, in call_at
    self._check_closed()
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
Call stack:
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 1770, in call_exception_handler
    self.default_exception_handler(context)
  File "/opt/conda/lib/python3.10/asyncio/base_events.py", line 1744, in default_exception_handler
    logger.error('\n'.join(log_lines), exc_info=exc_info)
Message: "Task was destroyed but it is pending!\ntask: <Task pending name='Task-1' coro=<Kernel.poll_control_queue() running at /opt/conda/lib/python3.10/site-packages/ipykernel/kernelbase.py:284> wait_for=<Future pending cb=[Task.task_wakeup()]> cb=[_chain_future.<locals>._call_set_state() at /opt/conda/lib/python3.10/asyncio/futures.py:391]>"
Arguments: ()

Comment (fenglin-ssb) [reply docker/jupyterlab/Dockerfile]: the error seems not related with labextensions, but widget.
should we drop this extension?

Comment (bjornandre) [reply docker/jupyterlab/Dockerfile]: It seems like a known issue that can be ignored for now: https://github.com/jupyter-widgets/ipywidgets/issues/3454

Comment (fenglin-ssb) [reply docker/jupyterlab/Dockerfile]: fixed. i delete jupyter-widgets, and after talking with Øyvind, the extension now works.
if a matrix variable can not be shown in a gridview.
add the following line `import pandas`

https://github.com/lckr/jupyterlab-variableInspector/issues/60#issuecomment-817235080

**Pull request merged**